Engine/GraphicsCore.cs

```diff
@@ -240,90 +204,80 @@ public static void Update()
 
         private static void RenderShadows()
         {
-            if (EngineCore.CurrentScene == null)
+            if (Scene.CurrentScene == null)
                 return;
 
             CurrentDevice.ImmediateContext.Rasterizer.State = frontCullingRasterizer;
             CurrentDevice.ImmediateContext.OutputMerger.BlendState = null;
 
-            List<GameObject> objects = EngineCore.CurrentScene.objects;
+            // List<GameObject> objects = EngineCore.CurrentScene.Objects;
 
             ShaderPipeline pipeline = null;
 
             void renderObjects()
             {
-                foreach (GameObject obj in objects)
+                foreach (MeshComponent meshComponent in Scene.FindComponentsOfType<MeshComponent>())
```

Comment:
add todo to test performance of manual objects iteration (allowing skipping of non-localEnabled objects) vs this implementation (that goes up the gameobject tree for every component)

Reply:
Use components LocalEnabled instead of Enabled